src/twin/mod.rs

```diff
@@ -0,0 +1,25 @@
+mod substrate_twindb;
+
+use crate::types::Message;
+use anyhow::{Ok, Result};
+use async_trait::async_trait;
+use parity_scale_codec::Decode;
+pub use substrate_twindb::*;
+
+#[async_trait]
+pub trait TwinDB {
+    async fn get(&self, twin_id: u32) -> Result<Twin>;
+}
+
+#[derive(Decode)]
```

Comment:
For this to be usable with the Cache changes as suggested above, this need to derive also the serder `Serialize` and `Deserialize` traits